iree/test/e2e/models/mnist_inference.mlir

```diff
@@ -0,0 +1,122 @@
+// MNIST model with empty weights, for testing.
+//
+// Generated by the colab/mnist_training.ipynb notebook with manual edits to
+// * remove remaining TensorFlow attributes
+// * replace large values with zeros
+//
+// Note: batch size is set to 32
+
+// RUN: iree-run-mlir -iree-hal-target-backends=vmvx %s -function-input="32x28x28x1xf32" | IreeFileCheck %s
+// RUN: [[ $IREE_LLVMAOT_DISABLE == 1 ]] || (iree-run-mlir -iree-hal-target-backends=dylib-llvm-aot %s -function-input="32x28x28x1xf32" | IreeFileCheck %s)
+// RUN: [[ $IREE_VULKAN_DISABLE == 1 ]] || (iree-run-mlir -iree-hal-target-backends=vulkan-spirv %s -function-input="32x28x28x1xf32" | IreeFileCheck %s)
+
+#map0 = affine_map<(d0, d1) -> (d0, d1)>
+#map1 = affine_map<(d0, d1) -> ()>
+#map2 = affine_map<(d0, d1) -> (d0)>
+#map3 = affine_map<(d0, d1) -> (d0, 0)>
+module  {
+  util.global private @"__iree_flow___sm_node22__model.layer-2.kernel" = dense<0.000000e+00> : tensor<784x128xf32>
```

Comment:
were these 0's in the model or something real? if real, we should make them non-zero splats so that it's easier to distinguish them through the pipeline (1.1, 1.2, 1.3, etc so they don't all fold together)